src/wcsnlen.c

```diff
@@ -0,0 +1,31 @@
+
+/*
+ * Copyright (c) 2018 Ken Cunningham
+ *
```

Comment:
I co-opted this right out of musl, so if you want to replace it with yours go right ahead. Otherwise we should credit them (MIT license).

Reply:
Ah, [I see](https://git.musl-libc.org/cgit/musl/tree/src/string/wcsnlen.c). I'll work in [their text](https://git.musl-libc.org/cgit/musl/tree/COPYRIGHT).